### PR TITLE
allow for empty VP when registering with discovery service

### PR DESCRIPTION
**Pull request by woutslakhorst**
fixes: #2916 

added simple check to see if to VCs are required

### CODE REVIEW

File: discovery/client.go [hunk 1]
```diff
@@ -137,7 +137,7 @@ func (r *defaultClientRegistrationManager) findCredentialsAndBuildPresentation(c
 	if err != nil {
 		return nil, fmt.Errorf("failed to match Discovery Service's Presentation Definition (service=%s, did=%s): %w", service.ID, subjectDID, err)
 	}
-	if len(matchingCredentials) == 0 {
+	if len(matchingCredentials) == 0 && len(service.PresentationDefinition.InputDescriptors) > 0 {
```
Comment (reinkrul) [inline discovery/client.go:140]: they could be optional? (`min: 0`)

Not sure if we already need to support this, though... Use case?

**Review by reinkrul — CHANGES_REQUESTED** [review]

**Review by gerardsn — COMMENTED** [review]
Isn't the real issue here that `PresentationDefinition.Match` doesn't return an error if it cannot fulfill the presentations definition? I think we allowed this because we initially thought a PD could be fulfilled by multiple wallets, but this is no longer the case?

File: discovery/client.go (hunk 1 above)
Comment (woutslakhorst) [reply discovery/client.go:140]: this is the check we do now on multiple locations.

**Comment by woutslakhorst** [comment]
> Isn't the real issue here that `PresentationDefinition.Match` doesn't return an error if it cannot fulfill the presentations definition? I think we allowed this because we initially thought a PD could be fulfilled by multiple wallets, but this is no longer the case?

uncertain for now, the spec allows for multiple wallets to provide the proof. When we return an error, we block that option. In our case we know we can't combine a organisation wallet with a user wallet.

**Comment by reinkrul** [comment]
Could be fixed with a specific error that can be tested? Since from PEX API intention perspective, not being able to fulfill a request (don't have the required credentials) is really different from fulfilling an empty request. It's now not possible to differentiate those 2.

**Comment by woutslakhorst** [comment]
> Could be fixed with a specific error that can be tested? Since from PEX API intention perspective, not being able to fulfill a request (don't have the required credentials) is really different from fulfilling an empty request. It's now not possible to differentiate those 2 (which is the actual problem, i.m.o.)

PD matching would have to rewritten to operate on a set of wallets, a bit too much for now (given future of pex is not certain as well).

**Review by reinkrul — APPROVED** [review]

**Pull request merged**